spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -400,34 +430,65 @@ private void doIfTransaction(Context ctx, Consumer<ReactiveFirestoreResourceHold
 	}
 
 	private <T> Write createUpdateWrite(T entity) {
-		String documentResourceName = buildResourceName(entity);
-		Document document = getClassMapper().entityToDocument(entity, documentResourceName);
-		return Write.newBuilder()
-				.setUpdate(document)
-				.build();
+		ResourceName resourceName = buildResourceName(entity);
```

Comment:
I would avoid introducing the `ResourceName` class to avoid coupling the resource name with whether the `id` was generated or not. Also mentioned below; I think it's not worth the documentation burden.

Instead, maybe refactor so it's like:

```
Object idVal = getOrGenerateEntityId(entity);
String resourceName = buildResourceName(Class<?> entityType, Object idVal);
```